File: python/src/core/tgon.py
from core.ngon import NGon
from core.point2d import Point2D

from math import sqrt

class TGon(NGon):
	def __init__(self, p):
		self.n = 3
		self.p = p

	def __str__(self):
		st = f"TGon: "
		for i in range(len(self.p)):
			st += f"{str(self.p[i])}, "
		st = st[:len(st)-2]
		return st

	def square(self):
		ab = Point2D.sub(self.p[1], self.p[0]).abs()
		bc = Point2D.sub(self.p[2], self.p[1]).abs()
		ca = Point2D.sub(self.p[0], self.p[2]).abs()
		pp = (ab + bc + ca)/2
		return sqrt(pp * (pp-ab) * (pp-bc) * (pp - ca))

File: csharp/WorkOOP/TGon.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace WorkOOP
{
    // Реализация класса TGon – треугольник.
    internal class TGon : NGon
    {
        /*
         * TGon – конструктор, описывающий инициализацию объекта. 
         * 
         * На вход принимает массив точек в порядке обхода по треугольнику.
         */
        public TGon(Point2D[] p) : base(p) { }

        /*
         * Переопределение под треугольник функции square()
         */
        new public double square()
        {
            double ab = Point2D.sub(p[1], p[0]).abs(), 
                bc = Point2D.sub(p[2], p[1]).abs(),
                ca = Point2D.sub(p[0], p[2]).abs(),
                pp = (ab + bc + ca)/2;
            return Math.Sqrt(pp * (pp - ab)*(pp - bc)*(pp - ca));
        }
    }
}
